src/main/resources/templates/db_scripts/populate-rules.sql

```diff
@@ -0,0 +1,97 @@
+      "orderNo": 0,
+      "errMessageId": "password.length.invalid"
+    }'),
+('dc653de8-f0df-48ab-9630-13aacfe8e8f4', '{
+      "ruleId": "dc653de8-f0df-48ab-9630-13aacfe8e8f4",
+      "name": "alphabetical_letters",
+      "type": "RegExp",
+      "validationType": "Strong",
+      "state": "Enabled",
+      "moduleName": "mod-password-validator",
+      "expression": "(?=.*[a-z])(?=.*[A-Z]).+",
+      "description": "The password must contain both upper and lower case letters",
+      "orderNo": 1,
+      "errMessageId": "password.alphabetical.invalid"
+    }'),
+('3e3c53ae-73c2-4eba-9f09-f2c9a892c7a2', '{
+      "ruleId": "3e3c53ae-73c2-4eba-9f09-f2c9a892c7a2",
+      "name": "numeric_symbol",
+      "type": "RegExp",
+      "validationType": "Strong",
+      "state": "Enabled",
+      "moduleName": "mod-password-validator",
+      "expression": "(?=.*\\d).+",
+      "description": "The password must contain at least one numeric character",
+      "orderNo": 2,
+      "errMessageId": "password.number.invalid"
+    }'),
+('2e82f890-49e8-46fc-923d-644f33dc5c3f', '{
+      "ruleId": "2e82f890-49e8-46fc-923d-644f33dc5c3f",
+      "name": "special_character",
+      "type": "RegExp",
+      "validationType": "Strong",
+      "state": "Enabled",
+      "moduleName": "mod-password-validator",
+      "expression": "(?=.*[!\"#$%&''()*+,-./:;<=>?@\\[\\]^_`{|}~]).+",
+      "description": "The password must contain at least one special character",
+      "orderNo": 3,
+      "errMessageId": "password.specialCharacter.invalid"
+    }'),
+('2f390fa6-a2f8-4027-abaf-ee61952668bc', '{
+      "ruleId": "2f390fa6-a2f8-4027-abaf-ee61952668bc",
+      "name": "no_user_name",
+      "type": "RegExp",
+      "validationType": "Strong",
+      "state": "Enabled",
+      "moduleName": "mod-password-validator",
+      "expression": "^(?:(?!<USER_NAME>).)+$",
+      "description": "The password must not contain your username",
+      "orderNo": 4,
+      "errMessageId": "password.usernameDuplicate.invalid"
+    }'),
+('8d4a2124-8a54-4c49-84c8-36a8f7fc01a8', '{
+      "ruleId": "8d4a2124-8a54-4c49-84c8-36a8f7fc01a8",
+      "name": "keyboard_sequence",
+      "type": "RegExp",
+      "validationType": "Strong",
+      "state": "Enabled",
+      "moduleName": "mod-password-validator",
+      "expression": "^(?:(?!qwe)(?!asd)(?!zxc)(?!qaz)(?!zaq)(?!xsw)(?!wsx)(?!edc)(?!cde)(?!rfv)(?!vfr)(?!tgb)(?!bgt)(?!yhn)(?!nhy)(?!ujm)(?!mju)(?!ik,)(?!,ki)(?!ol.)(?!.lo)(?!p;/)(?!/;p)(?!123).)+$",
+      "description": "The password must contain at least one special character",
+      "orderNo": 5,
+      "errMessageId": "password.keyboardSequence.invalid"
+    }'),
+('98b961b4-16b8-4e62-a359-abf3805e16b0', '{
+      "ruleId": "98b961b4-16b8-4e62-a359-abf3805e16b0",
+      "name": "repeating_characters",
+      "type": "RegExp",
+      "validationType": "Strong",
+      "state": "Enabled",
+      "moduleName": "mod-password-validator",
+      "expression": "^(?:(.)(?!\\1))*$",
```

Comment:
This is a complicated regexp for a valid password. It is much easier to write the regexp for an invalid password.
Consider adding a boolean field matchIsValid and extending the program logic.
matchIsValid==true means: If the regexp matches the password is valid, if it doesn't match the password is invalid.
matchIsValid==false means: If the regexp matches the password is invalid, if it doesn't match the password is valid.

Reply:
@julianladisch Thank you for the review and I understand that from the first side the regex seems complicated and it will look easy to do it vise versa, but it was written that way for the reason - the idea that each rule should be validated in its own way looks more complicated and easy to make a mistake, that is why it was decided to have a unified check for all RegExps (if password does not match regex --> it is wrong)even if you needed to write more complicated expression.